module/VuFind/src/VuFind/Content/Covers/LocalFile.php

```diff
@@ -0,0 +1,150 @@
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class LocalFile extends \VuFind\Content\AbstractCover
+{
+
+    /**
+     * Base filepath for LocalFile
+     *
+     * @var string
+     */
+    protected $basePath;
+
+    /**
+     * Filename for LocalFile
+     *
+     * @var string
+     */
+    protected $fileName;
+
+    /**
+     * Image type for LocalFile
+     *
+     * @var string
+     */
+    protected $imageType;
+
+    /**
+     * Does this plugin support the provided ID array?
+     *
+     * @param array $ids IDs that will later be sent to load() -- see below.
+     *
+     * @return bool
+     */
+    public function supports($ids)
+    {
+        return true;
+    }
+
+    /**
+     * VuFind configuration
+     *
+     * @var Config
+     */
+    protected $config;
+
+    /**
+     * Constructor
+     *
+     * @param Config $config VuFind configuration
+     */
+    public function __construct($config)
+    {
+        $this->config = $config;
+    }
+
+    /**
+     * Get image location from local file storage.
+     *
+     * @param string $key  local file directory path
+     * @param string $size Size of image to load (small/medium/large)
+     * @param array  $ids  Associative array of identifiers (keys may include 'isbn'
+     * pointing to an ISBN object and 'issn' pointing to a string)
+     *
+     * @return string|bool
+     */
+    public function getUrl($key, $size, $ids)
+    {
+        //Get settings from config.ini file
+        $config = $this->config;
+        $basePath = isset($config->$key->filepath)
+            ? $config->$key->filepath : null;
+        $fileNameToken = isset($config->$key->filename)
+            ? $config->$key->filename : null;
+        $imageType = isset($config->LocalFile->imagetype)
+            ? $config->$key->imagetype : null;
+
+        if (!isset($basePath) || !isset($fileNameToken) || !isset($imageType)) {
+            return false;
+        }
+
+        // Check base path for $VUFIND env variables
+        $vufindHome =  getenv('VUFIND_HOME');
+        $vufindLocal =  getenv('VUFIND_LOCAL_DIR') 
+        $basePath = str_replace("\$VUFIND_HOME", $vufindHome, $basePath);
+        $basePath = str_replace("\$VUFIND_LOCAL_DIR", $vufindLocal, $basePath);
+
+        // Add slash to basePath if it doesn’t exist
+        if (substr($basePath, -1) != "/") {
+            $basePath = $basePath."/";
+        }
+
+        // match fileName to information in $ids array
+        if (isset($ids[$fileNameToken])) {
+            $fileName = $ids[$fileNameToken];
+        } else {
+            return false;
+        }
+        // strip punctuation from imageType
+        $imageType = strtolower(preg_replace("/(\.)(.*)/", "$0 --> $2", $imageType));
+
+        // validate that imageType is allowed
+        $allowed_imageTypes = [ "gif", "jpg", "jpeg", "png", "tif", "tiff" ];
```

Comment:
Is it worth validating file type here, or should we let the higher-level code deal with that?

If we are going to do validation, it might be better to use a function that actually checks whether the contents of the file are a valid image, rather than going by the extension. I can find the details if you're not already familiar with doing this -- that way, the user isn't forced to use a particular extension, and you can avoid problems with case sensitivity in the current code here.